### PR TITLE
Feature/register service worker i

### PR DESCRIPTION
**Pull request by Matt-Roden**
# Description drawing

- added in the correct files and code for service worker registration and initial caching
- ran npm build

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor feature (non-breaking change that modifies existing work)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [ ] local host 3000
- [ ] console.log()
- [ ] dev tools
- [ ] cypress testing

# Checklist:

- [ ] My code follows the style guidelines of this project/linted
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -2,16 +2,25 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="icon" type="images/png" href="./Images/pokeball.png" />
```
Comment (hayleyw7) [inline public/index.html:5]: Thanks for fixing the favicon issue. The new one looks much better.

File: public/index.html [hunk 2]
```diff
@@ -2,16 +2,25 @@
...
-    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <link rel="apple-touch-icon" type="images/png" href="./Images/favicon.png" />
+    <link rel="manifest" href="./manifest.json" />
     <title>React App</title>
   </head>
   <body>
-    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <noscript>You need to enable JavasScript to run this app.</noscript>
     <div id="root"></div>
+    <script>
```
Comment (hayleyw7) [inline public/index.html:16]: This blows my mind, and I need to watch 500 videos about this to figure out the details.

File: public/index.html (hunk 1 above)
Comment (joanafbrito) [inline public/index.html:5]: great Job with the img

File: public/index.html [hunk 3]
```diff
@@ -2,16 +2,25 @@
...
+  if('serviceWorker' in navigator) {
+    window.addEventListener('load', () => {
```
Comment (hayleyw7) [inline public/index.html:18]: makes sense

File: public/manifest.json [hunk 4]
```diff
@@ -3,19 +3,10 @@
...
-      "type": "image/png",
-      "sizes": "192x192"
-    },
-    {
-      "src": "logo512.png",
+      "src": "/Images/pokeball.png",
+      "sizes": "720x720",
       "type": "image/png",
-      "sizes": "512x512"
+      "purpose": "any maskable"
```
Comment (joanafbrito) [inline public/manifest.json:9]: this one was very intresting

File: src/serviceWorkerRegistration.js [hunk 5]
```diff
@@ -56,6 +59,7 @@ function registerValidSW(swUrl, config) {
   navigator.serviceWorker
     .register(swUrl)
     .then((registration) => {
+      console.log('hello world<><>6')
```
Comment (hayleyw7) [inline src/serviceWorkerRegistration.js:62]: good, classic tests!

**Review by hayleyw7 — APPROVED** [review]

File: public/serviceworker.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    caches.keys()
+      .then((cacheNames) => Promise.all(
+        cacheNames.map((cacheName) => {
+          if(!cacheWhitelist.includes(cacheName)) {
+            return caches.delete(cacheName)
+          }
+        })
+      ))
+  )
+})
```
Comment (joanafbrito) [inline public/serviceworker.js:42]: Thank you so much for bringing this to live in our project, it looks super cool

File: src/serviceWorkerRegistration.js [hunk 7]
```diff
@@ -33,6 +34,7 @@ export function register(config) {
       const swUrl = `${process.env.PUBLIC_URL}/service-worker.js`;
```
Comment (joanafbrito) [inline src/serviceWorkerRegistration.js:34]: This line of code has a deep meening for this project! I'm proud of what you were able to do here! 🙃

**Review by joanafbrito — APPROVED** [review]
This is super cool, thanks for doing that

**Pull request merged**